### ISSUE TITLE
Access the `.next` method only once as per https://github.com/tc39/ecma262/pull/988

### ISSUE THREAD

**Issue by Jamesernator** [tc39/proposal-async-iteration#116]
The tests for this seem to be already merged into Test262 ([here](https://github.com/tc39/test262/pull/1250) and [here](https://github.com/tc39/test262/pull/1292)) so the spec text here should probably also be changed to reflect the changed behavior.

**Comment by domenic** [comment]
Yep, that's the plan, just haven't had the time yet. It's a bit tricky since a lot of the underlying spec machinery has changed out from under this proposal.
